src/dune_rules/findlib/findlib.ml

```diff
@@ -169,18 +171,39 @@ let builtin_for_dune : Dune_package.t =
   ; files = []
   }
 
-type db =
-  { stdlib_dir : Path.t
-  ; paths : Path.t list
-  ; builtins : Meta.Simplified.t Package.Name.Map.t
-  ; root_packages :
-      (Package.Name.t, (Dune_package.t, Unavailable_reason.t) result) Table.t
-  ; lib_config : Lib_config.t
-  }
+module DB = struct
+  type t =
+    { stdlib_dir : Path.t
+    ; paths : Path.t list
+    ; builtins : Meta.Simplified.t Package.Name.Map.t
+    ; lib_config : Lib_config.t
+    }
+
+  let equal t { stdlib_dir; paths; builtins; lib_config } =
+    Path.equal t.stdlib_dir stdlib_dir
+    && List.equal Path.equal t.paths paths
+    && Package.Name.Map.equal ~equal:Meta.Simplified.equal t.builtins builtins
+    && Lib_config.equal t.lib_config lib_config
+
+  let hash { stdlib_dir; paths; builtins; lib_config } =
+    Poly.hash
+      ( Path.hash stdlib_dir
+      , List.hash Path.hash paths
+      , Package.Name.Map.to_list builtins
+        |> List.hash (fun (k, v) ->
+               Tuple.T2.hash Package.Name.hash Meta.Simplified.hash (k, v))
+      , Lib_config.hash lib_config )
+end
+
+type t =
+  DB.t
+  * ( DB.t * Package.Name.t
```

Comment:
Why does this need to be a function? Looking at `create`, it seems to me that it could just be `type t = DB.t`.

Reply:
Yeah, it's not necessary. Removed.